frontend/src/pages/modelServing/screens/global/utils.ts

```diff
@@ -12,18 +13,31 @@ export const getTokenDisplayName = (secret: SecretKind): string =>
 export const getInferenceServiceActiveModelState = (
   is: InferenceServiceKind,
 ): InferenceServiceModelState =>
-  <InferenceServiceModelState | undefined>is.status?.modelStatus?.states?.activeModelState ||
-  <InferenceServiceModelState | undefined>is.status?.modelStatus?.states?.targetModelState ||
+  (is.status &&
+    typeof is.status.modelStatus !== 'undefined' &&
+    typeof is.status.modelStatus.states !== 'undefined' &&
+    (asEnumMember(is.status.modelStatus.states.activeModelState, InferenceServiceModelState) ||
+      asEnumMember(is.status.modelStatus.states.targetModelState, InferenceServiceModelState))) ||
   InferenceServiceModelState.UNKNOWN;
 
 export const getInferenceServiceStatusMessage = (is: InferenceServiceKind): string => {
-  const activeModelState = is.status?.modelStatus?.states?.activeModelState;
-  const targetModelState = is.status?.modelStatus?.states?.targetModelState;
-
   const failedToLoad = InferenceServiceModelState.FAILED_TO_LOAD;
+  let activeModelState;
+  let targetModelState;
+  let lastFailureMessage;
+
+  if (
+    is.status &&
+    typeof is.status.modelStatus !== 'undefined' &&
+    typeof is.status.modelStatus.states !== 'undefined'
+  ) {
+    activeModelState = is.status.modelStatus.states.activeModelState;
+    targetModelState = is.status.modelStatus.states.targetModelState;
+    lastFailureMessage = is.status.modelStatus.lastFailureInfo?.message;
+  }
+
   const isFailedToLoad = activeModelState === failedToLoad || targetModelState === failedToLoad;
 
-  const lastFailureMessage = is.status?.modelStatus?.lastFailureInfo?.message;
```

Comment:
Why change this function?

Reply:
> Made more explicit type guarding due to these type-check tests:
> 
> ```
> src/pages/modelServing/screens/global/utils.ts:16:16 - error TS18048: 'is.status.modelStatus' is possibly 'undefined'.
> 
> 16   asEnumMember(is.status?.modelStatus.states.activeModelState, InferenceServiceModelState) ||
> ```

Was a side effect of this; I just reverted the change.